src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -283,6 +287,13 @@ def _try_specializing_pnad(
         new_pnad = self._create_new_pnad_with_params_and_add_effects(
             pnad, updated_params, updated_add_effects)
 
+        if check_datastore_change:
+            # If the new PNAD has a datastore size that's not the same
+            # as that of the original PNAD, then we've potentially lost some
+            # data by specializing, which might do harm!
+            if len(new_pnad.datastore) != len(pnad.datastore):
```

Comment:
for my own understanding, could we assert here that that the lengths are related via < or > if they are not equal?

Reply:
+1, i think we can `assert len(new_pnad.datastore) < len(pnad.datastore)` in here, since data can never be added to the datastore?